sys-apps/diskus/diskus-0.3.1.recipe

```diff
@@ -0,0 +1,210 @@
+CHECKSUM_SHA256_37="153ffa32fd170e9944f7e0838edf824a754ec4c1fc64746fcc9fe1f8fa602e5d"
+
+SOURCE_URI_38="https://static.crates.io/crates/stable_deref_trait/stable_deref_trait-1.1.1.crate#noarchive"
+CHECKSUM_SHA256_38="dba1a27d3efae4351c8051072d619e3ade2820635c3958d826bfea39d59b54c8"
+
+SOURCE_URI_39="https://static.crates.io/crates/strsim/strsim-0.7.0.crate#noarchive"
+CHECKSUM_SHA256_39="bb4f380125926a99e52bc279241539c018323fab05ad6368b56f93d9369ff550"
+
+SOURCE_URI_40="https://static.crates.io/crates/term_size/term_size-0.3.1.crate#noarchive"
+CHECKSUM_SHA256_40="9e5b9a66db815dcfd2da92db471106457082577c3c278d4138ab3e3b4e189327"
+
+SOURCE_URI_41="https://static.crates.io/crates/textwrap/textwrap-0.10.0.crate#noarchive"
+CHECKSUM_SHA256_41="307686869c93e71f94da64286f9a9524c0f308a9e1c87a583de8e9c9039ad3f6"
+
+SOURCE_URI_42="https://static.crates.io/crates/thread_local/thread_local-0.3.6.crate#noarchive"
+CHECKSUM_SHA256_42="c6b53e329000edc2b34dbe8545fd20e55a333362d0a321909685a19bd28c3f1b"
+
+SOURCE_URI_43="https://static.crates.io/crates/ucd-util/ucd-util-0.1.2.crate#noarchive"
+CHECKSUM_SHA256_43="d0f8bfa9ff0cadcd210129ad9d2c5f145c13e9ced3d3e5d948a6213487d52444"
+
+SOURCE_URI_44="https://static.crates.io/crates/unicode-width/unicode-width-0.1.5.crate#noarchive"
+CHECKSUM_SHA256_44="882386231c45df4700b275c7ff55b6f3698780a650026380e72dabe76fa46526"
+
+SOURCE_URI_45="https://static.crates.io/crates/unreachable/unreachable-1.0.0.crate#noarchive"
+CHECKSUM_SHA256_45="382810877fe448991dfc7f0dd6e3ae5d58088fd0ea5e35189655f84e6814fa56"
+
+SOURCE_URI_46="https://static.crates.io/crates/utf8-ranges/utf8-ranges-1.0.2.crate#noarchive"
+CHECKSUM_SHA256_46="796f7e48bef87609f7ade7e06495a87d5cd06c7866e6a5cbfceffc558a243737"
+
+SOURCE_URI_47="https://static.crates.io/crates/vec_map/vec_map-0.8.1.crate#noarchive"
+CHECKSUM_SHA256_47="05c78687fb1a80548ae3250346c3db86a80a7cdd77bda190189f2d0a0987c81a"
+
+SOURCE_URI_48="https://static.crates.io/crates/version_check/version_check-0.1.5.crate#noarchive"
+CHECKSUM_SHA256_48="914b1a6776c4c929a602fafd8bc742e06365d4bcbe48c30f9cca5824f70dc9dd"
+
+SOURCE_URI_49="https://static.crates.io/crates/void/void-1.0.2.crate#noarchive"
+CHECKSUM_SHA256_49="6a02e4885ed3bc0f2de90ea6dd45ebcbb66dacffe03547fadbb0eeae2770887d"
+
+SOURCE_URI_50="https://static.crates.io/crates/walkdir/walkdir-2.2.6.crate#noarchive"
+CHECKSUM_SHA256_50="0ffb549f212c31e19f3667c55a7f515b983a84aef10fd0a4d1f9c125425115f3"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+SECONDARY_ARCHITECTURES="x86"
+fi
+
+PROVIDES="
+	diskus$secondaryArchSuffix = $portVersion
+	cmd:diskus
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:cargo$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	"
+
+defineDebugInfoPackage diskus$secondaryArchSuffix \
+	$binDir/diskus
+
+BUILD()
+{
+	mkdir -p ~/.cargo/registry/cache/github.com-1ecc6299db9ec823
```

Comment:
I honestly don't know exactly what it corresponds to (maybe @nielx knows?), but it's probably a commit, and is not specific to the application. If it somehow changes, cargo should automatically fetch the dependencies itself to the new directory; all you'll have to do is update the directory name to it.

Reply:
The hash most likely corresponds with a git revision of the github.com repository.

Reply:
It hasn't changed in quite a while, though, and it doesn't seem to match anything I can find. I was able to use a different directory with Gentoo's method, but it seems to need to find OS-specific dependencies, even when it knows not to use them.

So either we don't change it, clone the index ourselves (both of which take quite a while) and point it to it, try to find another way, or include the dependencies specific to other operating systems as sources, which should be the fastest unless theres a way to prevent it in the first place and is what Gentoo is doing.